qiskit/algorithms/eigensolvers/numpy_eigensolver.py

```diff
@@ -0,0 +1,297 @@
+        wavefn: np.ndarray,
+        threshold: float = 1e-12,
+    ) -> ListOrDict[tuple[complex, complex]]:
+
+        values: ListOrDict[tuple[complex, complex]]
+
+        # As a list, aux_operators can contain None operators for which None values are returned.
+        # As a dict, the None operators in aux_operators have been dropped in compute_eigenvalues.
+        if isinstance(aux_operators, list):
+            values = [None] * len(aux_operators)
+            key_op_iterator = enumerate(aux_operators)
+        else:
+            values = {}
+            key_op_iterator = aux_operators.items()
+        for key, operator in key_op_iterator:
+            if operator is None:
+                continue
+            value = 0.0
+            if operator.coeff != 0:
+                mat = operator.to_spmatrix()
+                # Terra doesn't support sparse yet, so do the matmul directly if so
+                # This is necessary for the particle_hole and other chemistry tests because the
+                # pauli conversions are 2^12th large and will OOM error if not sparse.
+                if isinstance(mat, scisparse.spmatrix):
+                    value = mat.dot(wavefn).dot(np.conj(wavefn))
+                else:
+                    if isinstance(operator, PauliSumOp):
+                        value = (
+                            Statevector(wavefn).expectation_value(operator.primitive)
+                            * operator.coeff
+                        )
+                    else:
+                        value = Statevector(wavefn).expectation_value(operator)
+                value = value if np.abs(value) > threshold else 0.0
+            # The value gets wrapped into a tuple: (mean, standard deviation).
+            # Since this is an exact computation, the standard deviation is known to be zero.
+            values[key] = (value, 0.0)
+        return values
+
+    def compute_eigenvalues(
+        self,
+        operator: BaseOperator | PauliSumOp,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+    ) -> NumPyEigensolverResult:
+
+        super().compute_eigenvalues(operator, aux_operators)
+
+        if operator is None:
+            raise AlgorithmError("Operator was never provided")
+
+        self._check_set_k(operator)
+        zero_op = PauliSumOp.from_list([("I", 1)]).tensorpower(operator.num_qubits) * 0.0
+        if isinstance(aux_operators, list) and len(aux_operators) > 0:
+            # For some reason Chemistry passes aux_ops with 0 qubits and paulis sometimes.
+            aux_operators = [zero_op if op == 0 else op for op in aux_operators]
+        elif isinstance(aux_operators, dict) and len(aux_operators) > 0:
+            aux_operators = {
+                key: zero_op if op == 0 else op  # Convert zero values to zero operators
+                for key, op in aux_operators.items()
+                if op is not None  # Discard None values
```

Comment:
My recollection is that we decided to drop support for None from the new estimate_observables. It was a kinda of "hack' for chemistry in the past and Nature is now using dicts where key/value pairs can simply be dropped. In the interests of expediency for now perhaps its easier to leave the logic since there are more places in this algo that deal with this.